### PR TITLE
Add Shared VPC L7 Internal LB example

### PR DESCRIPTION
**Pull request by msampathkumar**
Pull Request if for adding a Terraform example. A HTTP internal Load balancer example in a Shared VPC Environment.

About Shared VPC
1. Host Project
2. A Service Project (01) for LB Front End
3. A Service Project (02) for LB Back End - MIG

A Test VM is created Service Project 02, to test if Front End reachable with curl.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/templates/terraform/examples/shared_vpc_internal_http_lb_with_mig_backend.tf.erb [hunk 1]
```diff
@@ -0,0 +1,245 @@
+# project names
+variable "projects" {
```
Comment (slevenick) [inline mmv1/templates/terraform/examples/shared_vpc_internal_http_lb_with_mig_backend.tf.erb:2]: How is this supposed to work when running the integration test? The projects will not belong to the user who is running the tests

**Review by slevenick — CHANGES_REQUESTED** [review]

File: mmv1/templates/terraform/examples/shared_vpc_internal_http_lb_with_mig_backend.tf.erb (hunk 1 above)
Comment (msampathkumar) [reply mmv1/templates/terraform/examples/shared_vpc_internal_http_lb_with_mig_backend.tf.erb:2]: Hi @slevenick I understand what you are suggesting. As you can understand, this is a multi-project code sample(Shared VPC LB - b/232285286). I see we have published a multi project sample at https://registry.terraform.io/providers/hashicorp/google/4.10.0/docs/resources/compute_shared_vpc_host_project 

Can you suggest any pointers for me?

**Comment by slevenick** [comment]
It looks like the vpc_host project is very old and handwritten, so we don't have a similar pattern for generating that example. I'm not sure what you're trying to accomplish, but making sure it passes the generated test is important

**Comment by slevenick** [comment]
> @slevenick generated test(VCRtests) shows as failing but I don't see any errors. Can you share logs please?

You can find the logs [here](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6477/artifacts/47497922-d1ce-4c05-bcde-97310b86fe65/recording/TestAccComputeForwardingRule_sharedVPCInternalHttpLbWithMigBackendExample.log)

**Comment by msampathkumar** [comment]
Hi @slevenick, I have tested this samples several times and it is working.

In the automated test logs `TestAccComputeInstance_soleTenantNodeAffinities `, I see error `"message": "The resource 'projects/ci-test-project-188019/zones/us-central1-a/instances/tf-test-soletenant-n6jjozxw7l' was not found"`

So, this is Shared VPC example and for the instances to be created, instances need to have `network-user` role in the Host project where this network is defined.

Any suggestions please?

File: mmv1/templates/terraform/examples/shared_vpc_internal_http_lb_with_mig_backend.tf.erb [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+  health_checks         = [google_compute_region_health_check.default.id]
+  backend {
+    group           = google_compute_region_instance_group_manager.default.instance_group
+    balancing_mode  = "UTILIZATION"
+    capacity_scaler = 1.0
+  }
+}
+
+# health check
+resource "google_compute_health_check" "default" {
```
Comment (slevenick) [inline mmv1/templates/terraform/examples/shared_vpc_internal_http_lb_with_mig_backend.tf.erb:125]: This is missing provider = google-beta which is why the test fails to run

**Review by slevenick — CHANGES_REQUESTED** [review]

**Comment by c2thorn** [comment]
Hi @msampathkumar 
Sorry for the timing here, but MM no longer supports CGC samples. I believe the "terraform-on-cgc" internal documentation has just been updated with the new workflow, but essentially you'll be making PR's against https://github.com/terraform-google-modules/terraform-docs-samples directly

**Comment by msampathkumar** [comment]
Hi @c2thorn Thank you for the update!

**Pull request closed without merging**